crates/red_knot_python_semantic/src/types/infer.rs

```diff
@@ -3344,6 +3344,10 @@ impl<'db> TypeInferenceBuilder<'db> {
                     };
                 }
             }
+
+            KnownFunction::Len => {
+                // TODO: Assert that the argument implements the `Sized` protocol (correctly).
```

Comment:
This probably requires calling `__len__` to get the `CallDunderLenOutcome` and then verify the `binding` on it.